File: solutions/python/leap/1/leap.py
def leap_year(year: int) -> bool:
    """
    Determine whether a given year is a leap year.
    :param year: int - the year to check
    :return: bool - True if the year is a leap year, False otherwise
    """
    if year % 400 == 0:
        return True
    elif year % 100 == 0:
        return False
    elif year % 4 == 0:
        return True
    else:
        return False

File: solutions/go/leap/1/leap.go
// leap package defines a function to check if an year is a leap year or not
package leap

// IsLeapYear return true if the received year is a leap year and false otherwise
func IsLeapYear(year int) bool {
	switch {
    case year % 400 == 0:
        return true
    case year % 100 == 0:
        return false
    case year % 4 == 0:
        return true
    default:
        return false
    }
}
